### ISSUE TITLE
Outliner mode

### ISSUE THREAD

**Issue by ajvsol** [etesync/etesync-notes#3]
I've used various productivity tools for many years and one thing I've learnt is that flat hierarchical notes tools like Evernote (which has just 3 levels) become less and less useful over time as you add more data to them. Outliners like Roam Research, Dynalist and WorkFlowy however allow for infinitely nested lists and thus become more useful over time, not less. 

Individual EteSync notes in this paradigm would have their note title be used as the bullet title ("Bullet 1" in the screenshot below) and the note body with all it's markdown would be the bullet body ("Notes area" in the screenshot). 

![](https://i.imgur.com/fJvHKSQ.png)

With Dynalist you can set a global default for where this bullet body is always shown, 1st line only or hide it. Plus you can set exceptions to this for certain bullets.

![](https://i.imgur.com/swLyMy5.png)

It would be great to see EteSync Notes allow for an infinite hierarchy, as it stands now I don't see myself the tool with this limitation.

**Comment by tasn** [comment]
I would like to add wiki-like linking between notes, which I guess achieves this?

**Comment by ajvsol** [comment]
Not quite, having a full outliner mode (e.g try [this demo here](https://dynalist.io/demo)) enables a few additional advantages:
- Hierarchical organisation (plus you can additionally have wiki-like linking for additional organisation)
- Drag-and-drop reorganising
- Makes it very  easy to navigate to one particular note within thousands just by a few clicks dialling down by category
- Extremely scalable (I use an outliner app with hundreds of thousands of bullets and I find it very easy to find the info I'm looking for always)

**Comment by tasn** [comment]
I suspect the best approach may be to just integrate well with other note taking applications. :)

**Comment by nicosmaris** [comment]
Probably that's a separate feature request which is much simpler and is inspired by Github which renders headings in README.md as anchors and that helps navigate faster, scroll less, organize long notes, etc.

Setting [rules](https://github.com/iamacup/react-native-markdown-display#rules) at react-native-markdown-display that you [already](https://github.com/etesync/etesync-notes/blob/312ffbe0b55a9f3818776838afc5822d5d7d4a95/package.json#L42) use, should be enough to create anchors for heading1 up to heading6. So anyone can create any kinds of linking like outliners manually if the names of the anchors follow a naming convention and take the responsibility that deleting a note might end up with broken links if used in other notes.

For example the 3rd heading2 with a text like textwhatever could be rendered as `<a href="#2.3">textwhatever</a>` and referred to as `https://notes.etesync.com/notebook/longnotebookID/note/longnoteID#2.3`
